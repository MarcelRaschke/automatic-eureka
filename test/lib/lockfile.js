const nock = require('nock')
const { getNewLockfile } = require('../../lib/lockfile')

describe('getNewLockfile', async () => {
  beforeEach(() => {
    jest.resetModules()
  })

  nock.disableNetConnect()
  nock.enableNetConnect('localhost:5984')
  const lock = '{"name":"greenkeeper","version":"1.0.0","lockfileVersion":1,"requires":true,"dependencies":{"jest": {"version": "22.4.2"}}}'

  test('with changed package-lock.json', async () => {
    const { getNewLockfile } = require('../../lib/lockfile')
    const packageJson = '{"name": "greenkeeper","devDependencies": {"jest": "^22.4.3"}}'
    const newLock = '{"name":"greenkeeper","version":"1.0.0","lockfileVersion":1,"requires":true,"dependencies":{"jest": {"version": "22.4.3"}}}'

    nock('http://localhost:1234')
      .post('/', (body) => {
        expect(typeof body.type).toBe('string')
        expect(typeof body.packageJson).toBe('string')
        expect(typeof body.lock).toBe('string')
        expect(body).toMatchSnapshot()
        return true
      })
      .reply(200, () => {
        return {
          ok: true,
          contents: newLock
        }
      })

    await getNewLockfile({ packageJson, lock, type: 'npm' })
  })

  test('with package-lock.json', async () => {
    const packageJson = '{"name": "greenkeeper","devDependencies": {"jest": "^22.4.2"}}'

    nock('http://localhost:1234')
      .post('/', (body) => {
        expect(typeof body.type).toBe('string')
        expect(typeof body.packageJson).toBe('string')
        expect(typeof body.lock).toBe('string')
        expect(body).toMatchSnapshot()
        return true
      })
      .reply(200, () => ({ ok: false }))

    await getNewLockfile({ packageJson, lock, type: 'npm' })
  })

  test('with package-lock.json with Network Error', async () => {
    const httpTraffic = nock('http://localhost:1234')
      .post('/', (body) => {
        return true
      })
      .replyWithError({ code: 'ETIMEDOUT' })
      .post('/', (body) => {
        return true
      })
      .reply(200, () => ({ ok: true }))

    const packageJson = '{"name": "greenkeeper","devDependencies": {"jest": "^22.4.2"}}'
    await getNewLockfile({ packageJson, lock, type: 'npm' })
    expect(httpTraffic.isDone()).toBeTruthy()
    expect(httpTraffic.pendingMocks().length).toEqual(0)
  })
})
