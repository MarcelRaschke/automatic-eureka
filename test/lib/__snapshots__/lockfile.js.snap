// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getNewLockfile with changed package-lock.json 1`] = `
Object {
  "lock": "{\\"name\\":\\"greenkeeper\\",\\"version\\":\\"1.0.0\\",\\"lockfileVersion\\":1,\\"requires\\":true,\\"dependencies\\":{\\"jest\\": {\\"version\\": \\"22.4.2\\"}}}",
  "packageJson": "{\\"name\\": \\"greenkeeper\\",\\"devDependencies\\": {\\"jest\\": \\"^22.4.3\\"}}",
  "type": "npm",
}
`;

exports[`getNewLockfile with package-lock.json 1`] = `
Object {
  "lock": "{\\"name\\":\\"greenkeeper\\",\\"version\\":\\"1.0.0\\",\\"lockfileVersion\\":1,\\"requires\\":true,\\"dependencies\\":{\\"jest\\": {\\"version\\": \\"22.4.2\\"}}}",
  "packageJson": "{\\"name\\": \\"greenkeeper\\",\\"devDependencies\\": {\\"jest\\": \\"^22.4.2\\"}}",
  "type": "npm",
}
`;
